project.php

```diff
@@ -1974,6 +1969,78 @@ function sr_echo_time_form($label, $min_days, $max_days, $default_days, $extend
     echo "</form>\n";
 }
 
+function echo_smoothreading_options($project)
+{
+    $smooth_dir = "$project->dir/smooth";
+    $smooth_url = "$project->url/smooth";
+    echo "<li class='list-head'>", _("Download a Smooth Reading file");
+    echo "<ul>";
+    echo_file_downloads(glob("$smooth_dir/*.txt"), $smooth_url);
+
+    // zipped htm(l) file
+    foreach(glob("$smooth_dir/*.zip") as $zip_file)
+    {
+        $base_name = basename($zip_file);
+        $text = sprintf(_('%s (html file including any images)'), $base_name);
+        echo_download_item($smooth_url, $zip_file, $base_name, $text);
+    }
+
+    // no space after bracket commas
+    echo_file_downloads(glob("$smooth_dir/*.{epub,mobi}", GLOB_BRACE), $smooth_url);
+
+    // original uploaded zip file
+    $file_base_name = $project->projectid . "_smooth_avail.zip";
+    $file = "$project->dir/$file_base_name";
+    if(file_exists($file))
+    {
+        echo_download_item($project->url, $file, $file_base_name, "$file_base_name (all formats)");
+    }
+    echo "</ul>";
+    echo "</li>";
+
+    $files = glob("$smooth_dir/*.{txt,htm,html}", GLOB_BRACE);
+    // if sr file uploaded before transition to this mode there will not be a smooth directory
+    if($files)
+    {
+        echo "<li class='list-head'>", _("Open in browser (for reading only -- no annotations can be made):");
+        echo "<ul>";
+        foreach($files as $file)
+        {
+            $file_base_name = basename($file);
+            $url = "$smooth_url/$file_base_name";
+            echo "<li>";
+            echo "<a href='$url' target='_blank'>", html_safe($file_base_name), "</a>";
```

Comment:
Do we want to force opening in a new window/tab? My impression was that in general, that was discouraged, and that the user should be the one deciding whether to open a file in the current window, or in a new window or tab (whereas with `target='_blank'`, the user does not even have the option of opening something in the active window/tab).

Reply:
I don't know.